benchmark/distinct_user_id/nlohmann_json_sax.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#ifdef SIMDJSON_COMPETITION_NLOHMANN_JSON
+
+#include "distinct_user_id.h"
+#include <string.h>
+namespace distinct_user_id {
+
+using json = nlohmann::json;
+
+struct nlohmann_json_sax {
```

Comment:
Note that you are making the assumption that user/id pairs only appear in the main array or in retweeted_status objects. This is probably fair but I recommend you add a remark somewhere for the astute reader.

Reply:
I assumed that "id" key for a user could only be in a user object (whether that is inside the main array or inside a retweeted status) since if not in a user object, an "id" key could also refer to the id of the tweet.

Reply:
@NicolasJiaxin That's the correct definition, but I think your code would still pick this:

```
{"user":{"mother":{"id":1}}}
```

wouldn't it?

It is not necessarily a problem.

Reply:
(It is not a problem because I think that users are always given an idea so what I describe probably won't happen...
Though you could have
```
{"user":{"mother":{"id":1},"id":321312}}
```
maybe? But we don't work for NASA.)

Reply:
Oh yes, I see what you were saying by main array now. I'll add a remark concerning that then.